include/circt/Dialect/HW/PortImplementation.h

```diff
@@ -48,6 +48,11 @@ struct PortInfo : public ModulePort {
   // run, this will reflect the exact name of the port as emitted to the Verilog
   // output.
   StringRef getVerilogName() const;
+
+  bool operator==(const PortInfo &other) const {
+    return this->name == other.name && this->type == other.type &&
```

Comment:
Hmm, this equality operator makes sense for the linker use-case (ignore attributes and location information) but does it make sense in general?
If not, it may be better to define it elsewhere, if reasonable, so that there isn't unexpected behavior that results.

(submitting pending comment from months ago, whoops)